backend/apps/emails/tasks.py

```diff
@@ -0,0 +1,43 @@
+import smtplib
+
+from apps.emails.models import EmailLetter
+from config.celery import app
+from django.conf import settings
+from django.core.mail import send_mail
+
+
+@app.task(bind=True, default_retry_delay=2 * 60)
+def send_emails(self, email_letter: EmailLetter, context: dict = None):
+    """Celery task to send emails for Candidates."""
+    if not context:
+        context = {}
+    candidates = (candidate for candidate in email_letter.recipients.all() if candidate.email)
+    template = email_letter.template
+
+    email_letter.mark_in_progress()
+
+    for candidate in candidates:
+        try:
+            send_mail(
+                subject=template.render_subject(
+                    context=context,
+                    candidate=candidate,
+                    vacancy=email_letter.vacancy,
+                    event=email_letter.event,
+                ),
+                html_message=template.render_body(
+                    context=context,
+                    candidate=candidate,
+                    vacancy=email_letter.vacancy,
+                    event=email_letter.event,
+                ),
+                message=None,
```

Comment:
You can add:
```
from django.utils.html import strip_tags
message=strip_tags(html_message)
```
I guess, `message` is used if it's not able to render `html_message` so we still have some meaningful text inside the sending email.

Reply:
That will attach them both. It uses EmailMultiAlternatives under the hood and if it succeeds, we will dublicate the message then...

Reply:
It works and doesn't duplicate via gmail backend. Checked.